tools/codegen/dest/register_dispatch_key.py

```diff
@@ -59,6 +58,8 @@ class RegisterDispatchKey:
     @method_with_native_function
     def __call__(self, f: Union[NativeFunctionsGroup, NativeFunction]) -> List[str]:
         if isinstance(f, NativeFunctionsGroup):
+            # Note: We call gen_structured() if the operator is marked structured, regardless of the backend.
+            # gen_structured() has special logic to handle auto-generated kernels.
             if f.structured:
```

Comment:
Here's an example of how moving structured to be backend-dependent is "weird". We have special handling to auto-generate kernels for Meta and CompositeExplicitAutograd kernels for structured operators. Because of that, the decision to call `gen_structured()` should depend on whether the *operator* is structured, not the backend. Kinda weird- maybe instead I could just move the special handling of those keys directly into this function?

Reply:
Yeah, this is an interesting modeling question. An alternative to unconditionally calling `gen_structured` here is to have some preprocessing pass where we stuff the BackendIndex with entries for auto-generated kernels. What is going on here is OK-ish, though, since `gen_structured` knows how to call back to `gen_unstructured` if necessary.

Reply:
I think that this might work, if we mark `DispatchKey.Meta` and `DispatchKey.Composite[Explicit|Implicit]Autograd` keys as structured - since structured kernel codegen sometimes tries to autogenerate kernels for those keys if they don't exist.

If you're ok with it I'm going to write that down as a TODO and try it in a PR later in the stack, since it looks like this PR and the corresponding changes at https://github.com/pytorch/xla/pull/2915 are finally passing CI.